### PR TITLE
Fix MatchingRefApiCompat issues in Serialization libraries

### LINKED ISSUE #27981: MatchingRefApiCompat issues in Serialization libraries
See baseline files
https://github.com/dotnet/corefx/blob/master/src/System.Xml.XmlSerializer/src/MatchingRefApiCompatBaseline.uap.txt
https://github.com/dotnet/corefx/blob/master/src/System.Xml.XmlSerializer/src/MatchingRefApiCompatBaseline.uapaot.txt
https://github.com/dotnet/corefx/blob/master/src/System.Xml.XmlSerializer/src/MatchingRefApiCompatBaseline.netcoreapp.txt
https://github.com/dotnet/corefx/blob/master/src/System.Runtime.Serialization.Xml/src/MatchingRefApiCompatBaseline.txt
https://github.com/dotnet/corefx/blob/master/src/System.Runtime.Serialization.Formatters/src/MatchingRefApiCompatBaseline.txt

We should do one of the following:
1. Expose the APIs publicly in the reference assembly
2. Make the APIs in the implementation internal
3. Add a comment to the baseline file why you cannot do either (1) or (2) and keep them baselined.

See PR https://github.com/dotnet/corefx/pull/27881/ for extra context.

### PR DESCRIPTION
**Pull request by yujayee**
Fixes #27981
@mconnew @huanwu @zhenlan

### CODE REVIEW

**Comment by mconnew** [comment]
Failures are in System.Net.Security.Tests and not related to this change.

**Comment by yujayee** [comment]
@dotnet-bot test OSX x64 Debug Build
@dotnet-bot test Windows x86 Release Build
@dotnet-bot test Linux arm Release Build
@dotnet-bot test Tizen armel Debug Build

File: src/System.Runtime.Serialization.Xml/src/MatchingRefApiCompatBaseline.txt [hunk 1]
```diff
@@ -1,3 +1,2 @@
 Compat issues with assembly System.Runtime.Serialization.Xml:
```
Comment (weshaggard) [inline src/System.Runtime.Serialization.Xml/src/MatchingRefApiCompatBaseline.txt]: Can you please delete the baseline file now that it is empty?

Comment (weshaggard) [reply src/System.Runtime.Serialization.Xml/src/MatchingRefApiCompatBaseline.txt]: Similarly for the other empty ones.

**Pull request merged**